### PR TITLE
op-node: Monitor static peers in a background process

### PR DESCRIPTION
**Pull request by mslipper**
Currently, connections with static peers are not re-established in the event of disconnection. This causes us to have to restart replicas whenever we restart the sequencer. This PR adds a background process to monitor static peers for connectedness, and automatically reconnect when necessary.

### CODE REVIEW

File: op-node/p2p/host.go [hunk 1]
```diff
@@ -44,6 +52,73 @@ func (e *extraHost) ConnectionManager() connmgr.ConnManager {
...
+	tick := time.NewTicker(time.Minute)
+	defer tick.Stop()
+
+	for {
+		select {
+		case <-tick.C:
+			ctx, cancel := context.WithTimeout(context.Background(), time.Second*30)
+			var wg sync.WaitGroup
+
+			e.log.Debug("polling static peers")
```
Comment (protolambda) [inline op-node/p2p/host.go:95]:
```suggestion
			e.log.Debug("polling static peers", "peers", len(e.staticPeers))
```

**Review by protolambda — APPROVED** [review]
LGTM. One nit about adding logging in the loop. It's useful to know how many peers are being checked.

**Comment by mslipper** [comment]
Nice, thanks. Accepted the suggestion.

File: op-node/p2p/host.go [hunk 2]
```diff
@@ -142,26 +217,28 @@ func (conf *Config) Host(log log.Logger, reporter metrics.Reporter) (host.Host,
...
+
+	out := &extraHost{
+		Host:        h,
+		connMgr:     connMngr,
+		log:         log,
+		staticPeers: staticPeers,
+		quitC:       make(chan struct{}),
+	}
+	out.initStaticPeers()
+	if len(conf.StaticPeers) > 0 {
```
Comment (trianglesphere) [inline op-node/p2p/host.go:238]: Can we add static peers via the command line? It seems like we may always want the background process.

Comment (protolambda) [reply op-node/p2p/host.go:238]: We can add them via command line on start-up, so it's safe as read-only data in the go routine. We have an RPC to connect/protect dynamically, but not yet a notion of adding static peers to this re-connect loop after startup. Can be added later if really necessary.

Comment (mslipper) [reply op-node/p2p/host.go:238]: You mean via RPC? You can add a peer there, but it's stored separately from "static" peers which are just the ones users specify on the CLI at startup.

I think we should do reconnection logic for those peers separately - some of them might be inbound peers, or peers we don't care about reconnecting to.

**Pull request merged**